### PR TITLE
[recipe,doc] feat: add diffusionnft for Qwen-Image with npu backend

### PR DESCRIPTION
**Pull request by Sky-Trigger**
## What does this PR do?

Add an NPU training script for DiffusionNFT Qwen-Image OCR LoRA training.

This PR includes:

1. `examples/diffusionnft_trainer/run_qwen_image_ocr_lora_npu.sh`
2. README updates for launching the DiffusionNFT OCR LoRA example on Huawei Ascend NPUs.

The NPU script follows the existing GPU DiffusionNFT OCR LoRA recipe and adapts the runtime configuration for a single-node 16-NPU setup.

Main NPU-specific settings include:

* `actor_rollout_ref.model.attn_backend='_native_npu'`
* `actor_rollout_ref.rollout.tensor_model_parallel_size=2`
* `reward.reward_model.rollout.tensor_model_parallel_size=4`
* `trainer.n_gpus_per_node=16`
* `actor_rollout_ref.rollout.agent.num_workers=8`
* `reward.num_workers=4`

## Test

Validated the DiffusionNFT Qwen-Image OCR LoRA training script on a single-node 16-NPU Ascend environment.

Actor training metrics:

屏幕截图 2026-06-12 111137

Reward / critic metrics:

屏幕截图 2026-06-12 111703

Run command:

```bash
bash examples/diffusionnft_trainer/run_qwen_image_ocr_lora_npu.sh
```

The script runs with:

* policy model: `Qwen/Qwen-Image`
* reward model: `Qwen/Qwen3-VL-8B-Instruct`
* training batch size: 24
* rollout group size: 16
* rollout TP: 2
* reward TP: 4
* image resolution: 512
* total training steps: 300

## API and Usage Example

```bash
bash examples/diffusionnft_trainer/run_qwen_image_ocr_lora_npu.sh
```

Hydra overrides can be passed after the command, for example:

```bash
bash examples/diffusionnft_trainer/run_qwen_image_ocr_lora_npu.sh trainer.total_training_steps=100
```

## Design & Code Changes

This PR only adds an example recipe and documentation.

No Python API, trainer logic, rollout logic, or model implementation is changed.

The NPU script keeps the same DiffusionNFT training flow as the existing GPU script:

* online rollout with `vllm_omni`
* `old` LoRA adapter as rollout policy
* `default` LoRA adapter as the updated policy
* visual generative reward model for OCR scoring
* direct-preference DiffusionNFT loss

## Checklist Before Submitting

* [x] Add NPU example script.
* [x] Update documentation.
* [x] Manually validate the script on a 16-NPU Ascend environment.
* [x] No Python API changes.
* [x] No trainer or rollout implementation changes.

### CODE REVIEW

File: examples/diffusionnft_trainer/README.md [hunk 1]
```diff
@@ -114,11 +130,12 @@ See the [Metrics Documentation](../../docs/start/metrics.md) for a full descript
 
 ## Performance
 
-> All experiments were conducted on *NVIDIA H200* GPUs using the OCR reward.
+> All experiments were conducted on *NVIDIA H200* GPUs using the OCR reward. NPU experiments use *16× Ascend NPUs*.
 
 | Script | Model | Algorithm | Hybrid Engine | # Cards | Reward Fn | # GPUs for Actor | # GPUs for Rollout | # GPUs for Async Reward | Batch Size | `rollout.n` | lr   | # Val Samples | Training Samples per Step | `ppo_micro_batch_size_per_gpu` | Throughput (Samples / GPU / Seconds) | Time per Step (Seconds) |
```
Comment (AndyZhou952) [inline examples/diffusionnft_trainer/README.md]: Edit the table header `# GPUs for Actor` maybe to `# Cards for Actor` to include the NPU experiment result

File: examples/diffusionnft_trainer/README.md [hunk 2]
```diff
@@ -114,11 +130,12 @@ See the [Metrics Documentation](../../docs/start/metrics.md) for a full descript
...
 | --- | --- | --- | --- | --- | --- | --- | --- |-------------------------| --- | --- |------| --- | --- | --- |------------------------------| --------------------------------|
-| `run_qwen_image_ocr_lora.sh` | Qwen-Image | DiffusionNFT | True | 4 | qwenvl-ocr-vllm | 4 | 4 | 0 (sync)                | 24 | 16 | 3e-4 | 1k (full set) | 24×16=384 | 12 | 0.166                        | 570 |
+| `run_qwen_image_ocr_lora.sh` | Qwen-Image | DiffusionNFT | True | 4 (NVIDIA) | qwenvl-ocr-vllm | 4 | 4 | 0 (sync)                | 24 | 16 | 3e-4 | 1k (full set) | 24×16=384 | 12 | 0.166                        | 570 |
+| `run_qwen_image_ocr_lora_npu.sh` | Qwen-Image | DiffusionNFT | True | 16 (NPU) | qwenvl-ocr-vllm | 16 | 16 | 0 (sync)               | 24 | 16 | 3e-4 | 1k (full set) | 24×16=384 | 12 | 0.0078                      | ~490 |
```
[the remark this answers is not in the document]
Comment (AndyZhou952) [reply examples/diffusionnft_trainer/README.md]: Agree. Can double check the calculation?

File: examples/diffusionnft_trainer/run_qwen_image_ocr_lora_npu.sh [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+    trainer.project_name=diffusion_nft \
+    trainer.experiment_name=qwen_image_ocr_lora \
+    trainer.log_val_generations=8 \
+    trainer.val_before_train=False \
+    trainer.n_gpus_per_node=$NUM_GPUS_ACTOR_ROLLOUT_REWARD \
+    trainer.nnodes=1 \
+    trainer.save_freq=60 \
+    trainer.test_freq=20 \
+    trainer.total_epochs=1 \
+    trainer.total_training_steps=300 "$@"
```
Comment (AndyZhou952) [inline examples/diffusionnft_trainer/run_qwen_image_ocr_lora_npu.sh]: Enter for an extra line at the end of the script

File: examples/diffusionnft_trainer/README.md [hunk 4]
```diff
@@ -133,4 +150,5 @@ See the [Metrics Documentation](../../docs/start/metrics.md) for a full descript
   </tr>
 </table>
 
+
```
Comment (AndyZhou952) [inline examples/diffusionnft_trainer/README.md]:
```suggestion
```

**Review by AndyZhou952 — COMMENTED** [review]
Overall looks good. Just a few comments

File: examples/diffusionnft_trainer/README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (Sky-Trigger) [reply examples/diffusionnft_trainer/README.md]: Addressed the review comments, thanks.

File: examples/diffusionnft_trainer/README.md (hunk 1 above)
Comment (Sky-Trigger) [reply examples/diffusionnft_trainer/README.md]: Thanks, I’ve addressed the review comments.

File: examples/diffusionnft_trainer/run_qwen_image_ocr_lora_npu.sh (hunk 3 above)
Comment (Sky-Trigger) [reply examples/diffusionnft_trainer/run_qwen_image_ocr_lora_npu.sh]: Thanks, I’ve addressed the review comments.

**Review by AndyZhou952 — APPROVED** [review]
